CMakeLists.txt

```diff
@@ -96,6 +96,41 @@ if (WITH_MPFR)
     set(HAVE_SYMENGINE_MPFR yes)
 endif()
 
+# Piranha
+set(WITH_PIRANHA no
+    CACHE BOOL "Build with Piranha")
+
+# Boost
+set(WITH_BOOST no
+    CACHE BOOL "Build with Boost")
+
+# PTHREAD
+set(WITH_PTHREAD no
+    CACHE BOOL "Build with PTHREAD")
+
+if (WITH_PIRANHA)
+    set(WITH_BOOST yes)
+    set(WITH_PTHREAD yes)
+
+    find_package(Piranha REQUIRED)
+    include_directories(${PIRANHA_INCLUDE_DIRS})
+    set(HAVE_SYMENGINE_PIRANHA yes)
+endif()
+
+if (WITH_PTHREAD)
+    find_package(PTHREAD REQUIRED)
+    include_directories(${PTHREAD_INCLUDE_DIRS})
+    set(LIBS ${LIBS} ${PTHREAD_LIBRARIES})
+    set(HAVE_SYMENGINE_PTHREAD yes)
+endif()
+
+if (WITH_BOOST)
+    find_package(Boost REQUIRED COMPONENTS date_time filesystem system)
```

Comment:
I don't have much idea of finding COMMON_DIR, I'll open an issue so that we can keep track of this.